src/Microsoft.SqlTools.ServiceLayer/LanguageServices/PeekDefinition.cs

```diff
@@ -0,0 +1,210 @@
+                            End = new Position { Line = lineNumber + 1, Character = 1}
+                        }
+                    }
+            };
+            return locations;
+        }
+
+        /// <summary>
+        /// Get line number for the create statement
+        /// </summary>
+        private int GetStartOfCreate(string script, string createString)
+        {
+            string[] lines = script.Split(new string[] { Environment.NewLine }, StringSplitOptions.None);
+            for (int lineNumber = 0; lineNumber < lines.Length; lineNumber++)
+            {
+                if (lines[lineNumber].IndexOf( createString, StringComparison.OrdinalIgnoreCase) >= 0)
+                {
+                    return lineNumber;
+                }
+            }
+            return 0;
+        }
+
+        /// <summary>
+        /// Get the script of the selected token based on the type of the token
+        /// </summary>
+        /// <param name="declarationItems"></param>
+        /// <param name="tokenText"></param>
+        /// <param name="schemaName"></param>
+        /// <returns>Location object of the script file</returns>
+        internal Location[] GetScript(IEnumerable<Declaration> declarationItems, string tokenText, string schemaName)
+        {
+            foreach (Declaration declarationItem in declarationItems)
+            {
+                if (declarationItem.Title.Equals(tokenText))
+                {
+                    // Script object using SMO based on type
+                    DeclarationType type  = declarationItem.Type;
+                    if (sqlScriptGetters.ContainsKey(type))
+                    {
+                        return GetSqlObjectDefinition( 
+                                    sqlScriptGetters[type], 
+                                    tokenText, 
+                                    schemaName, 
+                                    sqlObjectTypes[type]
+                                ); 
+                    }
+                    return null;
+                }
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// Script a table using SMO
+        /// </summary>
+        /// <param name="tableName">Table name</param>
+        /// <param name="schemaName">Schema name</param>
+        /// <returns>String collection of scripts</returns>
+        internal  StringCollection GetTableScripts(string tableName, string schemaName)
+        {
+            return (schemaName != null) ? database.Tables[tableName, schemaName]?.Script()
+                    : database.Tables[tableName].Script();
+        }
+
+        /// <summary>
+        /// Script a view using SMO
+        /// </summary>
+        /// <param name="viewName">View name</param>
+        /// <param name="schemaName">Schema name </param>
+        /// <returns>String collection of scripts</returns>
+        internal StringCollection GetViewScripts(string viewName, string schemaName)
```

Comment:
>internal [](start = 8, length = 8)

Can't these be private now? You use the generic method in tests to get these so they no longer need internal accessibility.